resource/cmake/CommonFunctions.cmake

```diff
@@ -0,0 +1,443 @@
+    set(PROCESSORS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/processors")
+    set(TRANSFORMATIONS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/plugins/transformations")
+    set(EXAMPLES_DIR "${CMAKE_CURRENT_SOURCE_DIR}/examples")
+
+    set(STAGING_LIB_DIR "lib")
+    set(STAGING_BIN_DIR "bin")
+    set(STAGING_DOC_DIR "doc")
+    set(STAGING_TEST_DIR "test")
+    set(STAGING_EXAMPLES_DIR "examples")
+    set(STAGING_RESOURCE_DIR "resource")
+    set(STAGING_INCLUDE_DIR "include")
+
+    # If the RTIGATEWAY_ENABLE_${PluginName} is not specified, it will take the
+    # value of the RTIGATEWAY_ENABLE_ALL that is ON by default
+    option(RTIGATEWAY_ENABLE_ALL "Build everything" ON)
+    option(RTIGATEWAY_ENABLE_MODBUS "Build Modbus Adapter" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_MQTT "Build MQTT Adapter" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_FWD "Build Forwarder Processor" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_TSFM_FIELD "Build Field Transformation" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_TSFM_JSON "Build JSON Transformation" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_TESTS "Build tester applications for enabled plugins" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_EXAMPLES "Build examples applications for enabled plugins" ${RTIGATEWAY_ENABLE_ALL})
+    option(RTIGATEWAY_ENABLE_DOCS "Build documentation for enabled plugins" OFF)
+    option(RTIGATEWAY_ENABLE_SSL "Enable support for SSL/TLS" OFF)
+    option(RTIGATEWAY_ENABLE_LOG "Enable logging to stdout" OFF)
+    option(RTIGATEWAY_ENABLE_TRACE "Enable support for trace-level logging" OFF)
+
+    m_init_install_path()
+
+    # Set RTIGATEWAY_CONFIGURE_CONNEXT to check whether we need to configure
+    # RTI Connext DDS.
+    set(RTIGATEWAY_CONFIGURE_CONNEXT OFF)
+    if (RTIGATEWAY_ENABLE_MODBUS
+            OR RTIGATEWAY_ENABLE_MQTT
+            OR RTIGATEWAY_ENABLE_FWD
+            OR RTIGATEWAY_ENABLE_TSFM_FIELD
+            OR RTIGATEWAY_ENABLE_TSFM_JSON)
+        set(RTIGATEWAY_CONFIGURE_CONNEXT ON)
+    endif()
+
+    if (RTIGATEWAY_ENABLE_TESTS)
+        enable_testing()
+    endif()
+
+endmacro()
+
+#[[
+m_configure_connext_dds
+---------------------
+
+ * Type: macro
+ * Brief: configures RTI Connext DDS by using its FindPackage
+ * Params:
+ ** _CONNEXTDDS_VERSION: version of RTI Connext DDS to configure
+ * Restrictions: CONNEXTDDS_DIR should be set
+ * How to use it:
+
+    m_configure_connext_dds(6.1.0)
+]]
+
+macro(m_configure_connext_dds _CONNEXTDDS_VERSION)
+    # Configure Connext DDS dependencies
+    set_required_variable(CONNEXTDDS_DIR NDDSHOME CONNEXTDDS_DIR)
+
+    # CONNEXTDDS_ARCH is not required, but recommended
+    # eg -DCONNEXTDDS_ARCH=x64Linux4gcc7.3.0
+    if (DEFINED CONNEXTDDS_ARCH)
+        message(STATUS "CONNEXTDDS_ARCH set to <${CONNEXTDDS_ARCH}>")
+    endif()
+
+    list(APPEND CMAKE_MODULE_PATH "${CONNEXTDDS_DIR}/resource/cmake")
```

Comment:
Actually, that's the case. This is prepared, although not documented, to add another location for the findpackage, this will be useful for instance, when integrating the rticonnextdds-cmake-utils repo (which contains a FindPackage).